NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+                return availableSkills;
+            })
+    }
+
+    async sendCreateStationRequest(
+        agentStationGroupId: any,
+        subAccountId: any,
+        stationExtension: { toString: () => any },
+        agentUsername: string) {
+
+        let securityCode = this.generateSecurityCode(stationExtension);
+
+        let station = {
+            "agentStationGroupId": agentStationGroupId,
+            "clientId": subAccountId,
+            "extension": stationExtension,
+            "name": STATION_NAME,
+            "securityCode": securityCode,
+            "username": agentUsername
+        };
+
+        return this.session.post('/spokenAbc/jobs/stations', station)
+            .then((result: any) => {
+                // console.log(result.data)
+                return result
+            })
+    }
+
+    async getAgent(agentUsername: string) {
+        await this.session.login();
+
+        let agent = {};
+        try {
+            agent = await this.getAgentByUsername(agentUsername)
+        } catch (e) {
+            if (e.response.status === 404) {
+                console.log('agent ' + agentUsername + ' not found')
+            } else {
+                throw e
+            }
+        }
+
+        let subAccountId = await this.getSubAccountId();
+        let station = {};
+        try {
+            station = await this.getStationOnly(subAccountId, agentUsername);
+            if (!station) {
+                console.log('station associated with ' + agentUsername + ' not found');
+                station = {}
+            }
+        } catch (e) {
+            if (e.response.status === 404) {
+                console.log('station associated with ' + agentUsername + ' not found')
+            } else {
+                throw e
+            }
+        }
+
+        return {'agent':agent, 'station': station}
+    }
+
+    async getSubAccountId() {
+        return this.session.getSubAccount()
+            .then((response: { id: any }) => {
+                return response.id
+            })
+    }
+
+    async getAgentByUsername(agent_username: string) {
+        return this.session.get('/spokenAbc/agents/username/' + agent_username)
+            .then((response: { data: any }) => {
+                // console.log(response.data)
+                return response.data
+            })
+    }
+
+    async getAgentByLoginId(loginId: string) {
+        return this.session.get('/spokenAbc/agents/loginId/' + loginId)
+            .then((response: { data: any }) => {
+                // console.log(response.data)
+                return response.data
+            })
+    }
+
+    async waitForAgentCreation(loginId: string) {
+        return new Promise((resolve, reject) => {
+            process.stdout.write("Creating agent.");
+            let counter = 0;
+            const intervalId = setInterval(async () => {
+                try {
+                    await this.getAgentByLoginId(loginId);
```

Comment:
under what conditions this call could fail?

Reply:
Connection timeout is of them .